src/dmd/expressionsem.d

```diff
@@ -2000,6 +2001,28 @@ private bool functionParameters(const ref Loc loc, Scope* sc,
                 auto t = (p.type.ty == Tvoid) ? p.type : arg.type;
                 arg = toDelegate(arg, t, sc);
             }
+            else if ((p.storageClass & (STC.scope_ | STC.return_)) == STC.scope_)
```

Comment:
isn't that equivalent to `p.storageClass & STC.scope_ != 0` ?

Reply:
It means scope is on and return is off

Reply:
Ah right.
probably deserving of a comment?

Reply:
It's a common pattern.